src/commands/convert/msft_secure2hdf.ts

```diff
@@ -0,0 +1,130 @@
+      secureScore: scoreDoc,
+      profiles: profiles,
+    }),
+    withRaw,
+  )
+  fs.writeFileSync(checkSuffix(output), JSON.stringify(converter.toHdf()))
+}
+
+export default class MsftSecure2HDF extends Command {
+  static usage = [
+    'convert msft_secure2hdf -p <secureScoreProfile-json> -r <secureScore-json> [-h]',
+    'convert msft_secure2hdf -t <azure-tenant-id> -a <app-id> -s <app-secret> [-h]',
+  ];
+
+  static description =
+    'Translate a Microsoft365 Secure Score report to a Heimdall Data Format JSON file';
+
+  static examples = [
+    'saf convert msft_secure2hdf -p secureScoreProfile.json -r secureScoreControlProfiles -o output-hdf-name.json',
+  ];
+
+  static flags = {
+    help: Flags.help({char: 'h'}),
+    inputProfiles: Flags.string({
+      char: 'p',
+      required: false,
+      description:
+        'Input Microsoft Graph API "GET /security/secureScoreControlProfiles" output JSON File',
+    }),
+    inputScoreDoc: Flags.string({
+      char: 'r',
+      required: false,
+      description:
+        'Input Microsoft Graph API "GET /security/secureScores" output JSON File',
+    }),
+    tenantId: Flags.string({
+      char: 't',
+      required: false,
+      description: 'Azure tenant ID',
+    }),
+    appId: Flags.string({
+      char: 'a',
+      required: false,
+      description: 'Azure application ID',
+    }),
+    appSecret: Flags.string({
+      char: 's',
+      required: false,
+      description: 'Azure application secret',
+    }),
+    output: Flags.string({
+      char: 'o',
+      required: true,
+      description: 'Output HDF JSON file',
+    }),
+    'with-raw': Flags.boolean({
+      char: 'w',
+      required: false,
+      description: 'Include raw input file in HDF JSON file',
+    }),
+  };
+
+  async run() {
+    const {flags} = await this.parse(MsftSecure2HDF)
+    let scoreDoc: SecureScore
+    let profilesDoc: { value: SecureScoreControlProfile[] }
+
+    if (
+      flags.inputProfiles !== undefined &&
+      flags.inputScoreDoc !== undefined
+    ) {
+      // load from pre-downloaded files
+      scoreDoc = JSON.parse(fs.readFileSync(flags.inputScoreDoc, 'utf8'))
+      profilesDoc = JSON.parse(fs.readFileSync(flags.inputProfiles, 'utf8'))
+      processInputs(scoreDoc, profilesDoc, flags.output, flags['with-raw'])
+    } else if (
+      flags.tenantId !== undefined &&
+      flags.appId !== undefined &&
+      flags.appSecret !== undefined
+    ) {
+      // attempt to use the Graph API to pull files
+      const tenantId = flags.tenantId
+      const appId = flags.appId
+      const appSecret = flags.appSecret
+      const creds = new ClientSecretCredential(tenantId, appId, appSecret)
+      const graphClientOpts: ClientOptions = {
+        authProvider: new TokenCredentialAuthenticationProvider(creds, {
+          scopes: ['https://graph.microsoft.com/.default'],
+        }),
+      }
+      const graphClient: Client = Client.initWithMiddleware(graphClientOpts);
```

Comment:
see the asff integration as an example, but is there a way to pass through certs or choose to use an insecure network connection?  if so, please expose those options as well.

Reply:
pushed an (untested) fix to this request.

https://learn.microsoft.com/en-us/answers/questions/1389920/(aadsts500021)-error-occurs-when-getting-access-to